### PR TITLE
The Witcher Support

### PR DESCRIPTION
**Pull request by mattiascibien**
With this pull request I have  added support for the original Witcher game.

At the moment it is possibile to install the mods from NM and they will be extracted in the data folder. It is not complete at the moment but I am opening this to start a discussion on how to improve this (in most cases mod will be under the Data folder, but other zips have a different structure and will not work)

I will also need to improve the code style and write more comments. If anyone that has the steam version of the witcher that wants to check if detection is working will give me a huge favour. 

Here is a the screen of a mod working when added manually.

### CODE REVIEW

**Comment by DuskDweller** [comment]
Thank you very much for the commit!
I was wondering, why's there a bunch of Witcher 2 related files in it ?

**Comment by mattiascibien** [comment]
@DuskDweller The Witcher 2 was my base for learning how to implement a new game. I guess they have been changed in terms of indentation or similar (by mistake). I will try to remove the changes..

EDIT: For the csproj files I guess it is due to me having a slightly more recent version of Visual Studio 2015. Should not create any issue in any case

**Comment by DuskDweller** [comment]
Yes please. Can you also provide me a link to a bunch of mods I can use to easily and quickly test this?

**Comment by mattiascibien** [comment]
My tests were done with [Sex Takes Time Mod](http://www.nexusmods.com/witcher/mods/330/). Unfortunately thought other mods use a different directory structure inside the Zip and therefore they end up in the wrong section (IE: [this](http://www.nexusmods.com/witcher/mods/725)).

I was wondering on how is the standard for checking this on NMM.

**Comment by mattiascibien** [comment]
@DuskDweller is there any problem if I squish some commits so the edits of Witcher 2 project do not appear?

**Comment by DuskDweller** [comment]
Sure 👍

**Comment by DuskDweller** [comment]
Check this function inside the GamebryoGameModeBase:
public override string GetModFormatAdjustedPath(IModFormat p_mftModFormat, string p_strPath, bool p_booIgnoreIfPresent)

This is how we handle archives unproperly packaged, as long as they have a "Data" folder in it.

**Comment by mattiascibien** [comment]
@DuskDweller thanks. I will have a look onto it.

File: NexusClient.sln [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 14
-VisualStudioVersion = 14.0.24720.0
```
Comment (mattiascibien) [inline NexusClient.sln:4]: This line should not cause problems even with old versions of VS2015 @DuskDweller

**Comment by mattiascibien** [comment]
@DuskDweller [Celina Wreath Fix](http://www.nexusmods.com/witcher/mods/741/?tab=2&navtag=http%3A%2F%2Fwww.nexusmods.com%2Fwitcher%2Fajax%2Fmodfiles%2F%3Fid%3D741&pUp=1) is another mod that should work. I am working on identifying various mods and I come to this:
- If it is a single folder -> extract into `data` (it seems that it is also possible to extract those inside Data\Override)
- If it has a `data` folder inside -> extract the folders inside that int `data`
- If it has an override folder -> extract into Data\Override

Seems that most of the content should be extracted inside the Data\Override folder. I'll check this.

**Comment by mattiascibien** [comment]
@DuskDweller I have moved everything inside Data\Override as it is intended for Aurora Engine and Derivate games.

**Comment by DuskDweller** [comment]
Hi @mattiascibien , how's it going with the Witcher support?
Let me know should you need further help.

**Comment by mattiascibien** [comment]
Hi @DuskDweller. At the moment I am trying to figure the best way to decompress and place in the right dir most of the mods included in the site. 

However I am not sure if [this](https://github.com/mattiascibien/Nexus-Mod-Manager/blob/436a19cfefa6d56b74720fd48a8c13f41a17aac7/Witcher/WitcherGameMode.cs#L71-L73) is the best way to do so.

I also need to have a look on how to work with ENBs and similars.

**Comment by DuskDweller** [comment]
Hey @mattiascibien sorry for the late reply, we currently don't allow ENBs and the likes (exe and dll files that go in the game's folder) to be installed through NMM for safety reasons, so you can ignore them for the time being.

**Comment by dyslexicanaboko** [comment]
@DuskDweller @mattiascibien - this is a really old pull request. What will happen with it?

**Comment by DuskDweller** [comment]
Closing this as it's been clearly abandoned.

**Pull request closed without merging**